projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -0,0 +1,231 @@
+
+  static nextId = 0;
+
+  form: FormGroup;
+  stateChanges = new Subject<void>();
+  @HostBinding() id = `kui-time-input-${TimeInputComponent.nextId++}`;
+  focused = false;
+  errorState = false;
+  controlType = 'kui-time-input';
+  matcher = new TimeInputErrorStateMatcher();
+  onChange = (_: any) => {};
+  onTouched = () => {};
+
+  @Input() dateLabel = 'date';
+  @Input() timeLabel = 'time';
+
+  dateFormControl: FormControl;
+
+  timeValidator = [Validators.pattern(CustomRegex.TIME_REGEX)];
+  timeFormControl: FormControl;
+
+  datePipe = new DatePipe('en-US');
+
+  get empty() {
+    const userInput = this.form.value;
+    return !userInput.date && !userInput.time;
+  }
+
+  @HostBinding('class.floating')
+  get shouldLabelFloat() {
+    return this.focused || !this.empty;
+  }
+
+  @Input()
+  get required() {
+    return this._required;
+  }
+
+  set required(req) {
+    this._required = coerceBooleanProperty(req);
+    this.stateChanges.next();
+  }
+  
+  private _required = false;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+    this._disabled ? this.form.disable() : this.form.enable();
+    this.stateChanges.next();
+  }
+
+  private _disabled = false;
+
+  @Input()
+  get placeholder() {
+    return this._placeholder;
+  }
+
+  set placeholder(plh) {
+    this._placeholder = plh;
+    this.stateChanges.next();
+  }
+
+  private _placeholder: string;
+
+  @Input() readonly = false;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  @Input()
+  get value(): string | null {
+    const userInput = this.form.value;
+    if (userInput.date !== null && userInput.time !== null) {
+      return this.userInputToTimestamp(userInput);
+    }
+    return null;
+  }
+
+  set value(timestamp: string | null) {
+    if (timestamp !== null) {
+      const dateTime = this.convertTimestampToDateTime(timestamp);
+      this.form.setValue({date: dateTime[0], time: dateTime[1]});
+    } else {
+      this.form.setValue({date: null, time: null});
+    }
+    this.stateChanges.next();
+  }
+
+  @Input() errorStateMatcher: ErrorStateMatcher;
+
+  constructor(fb: FormBuilder,
+              @Optional() @Self() public ngControl: NgControl,
+              private fm: FocusMonitor,
+              private elRef: ElementRef<HTMLElement>,
+              @Optional() _parentForm: NgForm,
+              @Optional() _parentFormGroup: FormGroupDirective,
+              _defaultErrorStateMatcher: ErrorStateMatcher) { 
+
+    super(_defaultErrorStateMatcher, _parentForm, _parentFormGroup, ngControl);
+
+    this.dateFormControl = new FormControl(null);
+    this.dateFormControl.setValidators([Validators.required]);
```

Comment:
I think you can use the more compact form like below to set the validator.

Reply:
fixed in bfec41b